pipeline/test_pipeline.py

```diff
@@ -0,0 +1,64 @@
+"""Tests for extracting raw data from API, transforming and loading into database."""
+import unittest
+from unittest.mock import patch
+import requests
+from extract import get_plant_data, parse_plant_data, extract_botanist_name
+
+
+class TestExtractScript(unittest.TestCase):
+    """Tests for the extract portion of the pipeline."""
+
+    @patch("extract.requests.get")
+    def test_get_plant_data_success(self, mock_get):
+        """Test get_plant_data successful."""
+        mock_get.return_value.status_code = 200
+        mock_get.return_value.json.return_value = {
+            "plant_id": 1, "name": "Test Plant"}
+
+        result = get_plant_data(1)
+        self.assertEqual(result["plant_id"], 1)
+        self.assertEqual(result["name"], "Test Plant")
+
+    @patch("extract.requests.get")
+    def test_get_plant_data_unsuccessful(self, mock_get):
+        """Test get_plant_data with an unsuccessful API response."""
+        mock_get.return_value.status_code = 404
+        mock_get.side_effect = requests.RequestException("API not found")
+
+        result = get_plant_data(999)
+        self.assertIsNone(result)
+
```

Comment:
Could you add a test for when there are sensor errors? For example when the API shows this: 
```
{
  "error": "plant not found",
  "plant_id": 7
}
```